src/middlewared/middlewared/main.py

```diff
@@ -1370,8 +1370,19 @@ def terminate(self):
             # in base class to reduce number of awaits
             if hasattr(service, "terminate"):
                 self.logger.trace("Terminating %r", service)
+                terminate_timeout_func = getattr(service, 'terminate_timeout', None)
```

Comment:
This all should probably be within the try too, if something failes we dont bog down whole shutdown process.